Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/DscSignatureValidator.kt

```diff
@@ -143,6 +139,7 @@ class DscSignatureValidator @Inject constructor(
     }
 
     companion object {
+        private val certificateFactory = CertificateFactory.getInstance("X.509")
```

Comment:
We should make this a `lazy`.
It's tricky when `companion object` is initialized, and this call may access the provider, docs:

```
     * <p> Note that the list of registered providers may be retrieved via
     * the {@link Security#getProviders() Security.getProviders()} method.
```

We don't want this to accidentally initilize before the provider is setup (i.e. `Security.add` was called).

Sorry I only thought now of it.